cmd/soroban-rpc/internal/test/simulate_transaction_test.go

```diff
@@ -468,9 +467,9 @@ func TestSimulateInvokeContractTransactionSucceeds(t *testing.T) {
 	assert.Nil(t, obtainedAuth.RootInvocation.SubInvocations)
 
 	// check the events
-	assert.Len(t, response.Events, 1)
+	assert.Len(t, response.Events, 3)
 	var event xdr.DiagnosticEvent
-	err = xdr.SafeUnmarshalBase64(response.Events[0], &event)
+	err = xdr.SafeUnmarshalBase64(response.Events[1], &event)
```

Comment:
a followup on @sreuland remarks - if the contract now returns 3 events, shouldn't we test all of them ?
note that it shouldn't be a blocker for this PR; but we should note to ourselves to address that.

Reply:
The other events are debug events, we should test for the event emitted explicitly by the contract implementation